### ISSUE TITLE
Create fail_over Option

### ISSUE THREAD

**Issue by krisdestruction** [pytest-dev/pytest-cov#320]
We support `--cov-fail-under` and `fail_under` options to fail tests when tests are under some coverage amount. https://github.com/pytest-dev/pytest-cov/pull/141

In an effort to asymptotically get to 100%, some repos may want `--cov-fail-over` and `fail_over` options to ensure the `fail_under` config is updated.

**Comment by blueyed** [comment]
> some repos may want `--cov-fail-over` and `fail_over` options to ensure the `fail_under` config is updated.

Really?  So that you get the extra work of updating the lower limit in case you manage to improve coverage just a bit too much?

I rather suggest using services that a) ensure that the diff is covered enough, and b) that coverage does not drop in general, but only increases.

Basically you should use the current/previous percentage for `fail_under` always.  This could also be done by storing it somewhere as an artifact on CI, when not using services like codecov that manage it for you.

**Comment by krisdestruction** [comment]
Yes really, because working with legacy systems with lower coverage means that there's lots of room to improve. Can you give an example of such a service you're suggesting?

As for my use case, I can't imagine a fail_over would be too code divergent than a fail_under

**Comment by blueyed** [comment]
I suggest using https://codecov.io/ - it can provide status about the patch being covered, and also the project's coverage not dropping.
It defaults to having the patch covered by the amount of current coverage (IIRC), but can be made to check for it to be covered by e.g. 100%.

**Comment by ionelmc** [comment]
I think it's an opportune moment to start working on a hookspec so people
can implement any checks like fail over and stop the endless process of
adding cli options.

**Comment by nedbat** [comment]
I suggest there's no need to add this to pytest-cov at all.  Let pytest run the tests. Let coverage report on coverage. See #337.
